scipy/signal/_splinemodule.cc

```diff
@@ -201,49 +210,56 @@ static PyObject *IIRsymorder1_ic(PyObject *NPY_UNUSED(dummy), PyObject *args)
   const npy_intp sz[2] = {M, 1};
   dtype = PyArray_DescrFromType(thetype);
   out = (PyArrayObject *)PyArray_Empty(2, sz, dtype, 0);
-  if (out == NULL) goto fail;
+  if (out == NULL) {
+    // Duplicate the `goto fail` block to avoid "crosses initialization" build errors.
+    Py_XDECREF(a_sig);
+    Py_XDECREF(out);
+    return NULL;
+  }
 
   switch (thetype) {
   case NPY_FLOAT:
     {
       float rz1 = z1.real;
 
       if ((precision <= 0.0) || (precision > 1.0)) precision = 1e-6;
-      ret = S_SYM_IIR1_initial(rz1, (float *)PyArray_DATA(a_sig),
-                               (float *)PyArray_DATA(out), M, N,
-                               (float )precision);
+        ret = _sym_iir1_initial(rz1, static_cast<float *>(PyArray_DATA(a_sig)),
+                               static_cast<float *>(PyArray_DATA(out)), M, N,
+                               static_cast<float>(precision));
```

Comment:
Increasing the indentation here makes it seem like this is actually the `if` block; however, that's not the case.

Better to make this unambiguous:
```C
      if ((precision <= 0.0) || (precision > 1.0))
        precision = 1e-6;
                                                                               // <- intentional newline
      ret = _sym_iir1_initial(rz1, static_cast<float *>(PyArray_DATA(a_sig)),
                             static_cast<float *>(PyArray_DATA(out)), M, N,
                             static_cast<float>(precision));
```
(162ed2d602240bb291b0a927ea7f94426789b36b)